enhancements/installer/azurestack.md

```diff
@@ -0,0 +1,160 @@
+environments](azure-support-known-cloud-environments.md) and on-prem platforms. This enhancement
+covers details for the Installer to create infrastructure and run an OpenShift cluster on ASH.
+
+## Motivation
+
+Currently, Azure Stack Hub is not supported but is a significant on-prem platform.
+
+### Goals
+
+1. Installer creates infrastructure and installs OpenShift on Azure Stack Hub.
+2. All standard support that entails, such as destroy.
+
+### Non-Goals
+
+1. ASH will not "inherit" established Azure
+features such as using existing networks or resource groups.
+
+## Proposal
+
+
+### User Stories
+
+#### Story 1
+As a cluster admin, I want to be able to use `openshift-install` to create and destroy a cluster on
+Azure Stack Hub.
+
+#### Story 2
+As an OpenShift developer, I want Azure Stack Hub to be managed and released like other platforms and covered by CI tests.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Primary ways in which Azure Stack differs from Azure:
+- requires a different rhcos image
+- limited instance metadata service (IMDS) for VMs
+- does not support private DNS zones
+- limited subset of Azure infrastructure available (ergo different Terraform provider)
+
+### Risks and Mitigations
+
+Adding ASH support is somewhere in between adding a new Azure environment and an entirely new platform.
+
+## Design Details
+
+### Infrastructure Global Configuration
+
+#### API
+ASH will need to be added to the known [Azure Cloud Environments in the API](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L339-L355):
+
+```go
+// AzureStackCloud is the Azure cloud environment used on-premises.
+AzureStackCloud AzureCloudEnvironment = "AzureStackCloud"
+```
+
+#### Cluster Config Operator
+Cluster config operator will [need to be updated](https://github.com/openshift/cluster-config-operator/blob/master/pkg/operator/kube_cloud_config/azure.go#L23) and the new API vendored in.
+
+#### API endpoints
+ASH endpoints are user-provided and, therefore, the Azure SDK treats ASH endpoints differently than the already-known endpoints of other Azure environments. When the cloud environment is set to
+`AZURESTACKCLOUD` the SDK expects the environment variable `AZURE_ENVIRONMENT_FILEPATH` to point to a [json configuration file](https://kubernetes-sigs.github.io/cloud-provider-azure/install/configs/#azure-stack-configuration), which is [typically located at `/etc/kuberentes/azurestackcloud.json`](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/151).
+
+The Installer will need to construct this json configuration file from user input and include the file as part of the cloud-provider-config configmap. The file will also need to be present on nodes for the kubelet. `AZURE_ENVIRONMENT_FILEPATH` will need to be set programmatically for ASH on the kubelet, presumably through the kubelet's systemd unit [(see open questions)](#open-questions).
```

Comment:
Where will this user input come from? Is the expectation that they will have already set up an `AZURE_ENVIRONMENT_FILEPATH` on the install machine, given that the installer will presumably need to know the endpoints for its accesses to the Azure API?

Reply:
I think the update should clarify this.